examples/hangman.rs

```diff
@@ -0,0 +1,274 @@
+//! A hangman game to guess the hidden word.
+// To run this example, use the command: cargo run --features unstable --example hangman
+#[cfg(all(not(feature = "unstable")))]
+compile_error!("This example relies on unstable features. Run with `--features unstable`");
+
+use std::io::{Error, Write};
+
+use graphic::HangmanDesign;
+use turtle::{rand::RandomSlice, Drawing, Point, Speed, Turtle};
+
+// A small word list to choose from
+const WORDS: &[&str; 28] = &[
+    "poetry",
+    "formal",
+    "type",
+    "expresses",
+    "personal",
+    "emotions",
+    "feelings",
+    "typically",
+    "spoken",
+    "equivalent",
+    "lyrics",
+    "term",
+    "derives",
+    "literature",
+    "defined",
+    "musical",
+    "accompaniment",
+    "usually",
+    "stringed",
+    "instrument",
+    "term",
+    "importance",
+    "literary",
+    "theory",
+    "division",
+    "developed",
+    "between",
+    "categories",
+];
+
+fn main() {
+    let mut drawing = Drawing::new();
+
+    // the turtle that is responsible for drawing the hangman design
+    let mut hangman = drawing.add_turtle();
+
+    // the turtle that is responsible for drawing the lines on which the letters appear.
+    let mut lines = drawing.add_turtle();
+
+    // the turtle that is responsible for drawing the Won or lost smiley.
+    let mut wl = drawing.add_turtle();
+    wl.hide();
+
+    // the secret word that is to be guessed
+    let secret = *WORDS.choose().unwrap();
+
+    // this list stores all the steps of the hangman so that they can be drawn from a for loop or an index.
+    let hangman_design = HangmanDesign::default();
+    graphic::setup(&mut hangman);
+
+    let mut all_guesses = String::new();
+
+    // Loop over each step basically there are two ways to do this either one loops over the steps or one loops over guesses. Both need two nested loops.
+    // HangmanDesign has the [`Iterator`] trait implemented which is why we can loop it directly.
+    // The `'won:` syntax is to be able to conviniently `break` from both nested loops.
```

Comment:
Nice! These kinds of explanatory comments for less-frequently-used syntax are great. 😁